packages/react-app/src/lib/queryClient.ts

```diff
@@ -12,6 +13,9 @@ const queryClient = new QueryClient({
 });
 
 const GetGiveaways = () => useQuery('active', API.getGiveaways);
+const GetGiveawayDetails = (id?: string) => useQuery<Giveaway | undefined>(['details', id], () => id ? API.getGiveaway(id) : undefined);
```

Comment:
In this case we need the pascal case to avoid error "React Hook "useQuery" is called in function "getGiveawayDetails" that is neither a React function component nor a custom React Hook function."